src/configuration.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::{anyhow, Result};
+use crate::cli::CliOptions;
+use dirs::config_dir;
+use merge::Merge;
+use std::{fs::{self, create_dir, File}, path::Path};
+use std::io::{BufReader, Read};
+
+pub fn read_cfg_file<P: AsRef<Path>>(path: &P) -> Result<CliOptions> {
+    let file = File::open(&path)?;
+    let mut buf_reader = BufReader::new(file);
+    let mut contents = String::new();
+    buf_reader.read_to_string(&mut contents)?;
+    Ok(toml::from_str(contents.as_str()).unwrap())
+}
+
+pub fn load_cfg<P: AsRef<Path>>(path: Option<&P>) -> Result<CliOptions> {
+    match path {
+        Some(config_path) => read_cfg_file(config_path),
+        None => {
+            let mut default_path = config_dir().unwrap();
+            default_path.push("/onefetch/config.toml");
+            if default_path.exists() {
+                read_cfg_file(&default_path)
+            } else {
+                create_dir(&default_path)?;
+                write_default_cfg(&default_path);
+                Err(anyhow!("Configuration file at {:?} does not exist!", default_path))
+            }
+        }
+    }
+}
+
+pub fn write_default_cfg<P: AsRef<Path>>(default_path: &P) {
+    let toml = toml::to_string(&CliOptions::default()).expect("Config should be serializable");
+    fs::write(default_path, toml).expect("Unable to write file");
+}
```

Comment:
Looks mostly good, but I would move these to be implemented on an appropriate struct.
```rust
impl Config {
    fn load_from_file<P: AsRef<Path>>(path: P) -> Result<Self> {
        // ...
    }
}
```